api/handlers.go

```diff
@@ -92,22 +110,30 @@ func (api *API) loginGET(w http.ResponseWriter, req *http.Request, _ httprouter.
 
 // loginPOST starts a user session by issuing a cookie
 func (api *API) loginPOST(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	defer api.closeBody(req)
+	// Get the body, we might need to use it several times.
+	body, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		api.WriteError(w, errors.AddContext(err, "empty request body"), http.StatusBadRequest)
```

Comment:
`nit` context might be misleading, it might not be empty right?

Reply:
That's true. Fixing.